src/ruby/puppet-server-lib/puppet/server/certificate.rb

```diff
@@ -15,6 +16,7 @@ def initialize(java_cert)
     end
 
     @java_cert = java_cert
+    @oid_defns = Puppet::Server::Oids.load_custom_oid_file(Puppet[:trusted_oid_mapping_file]) + Puppet::SSL::Oids::PUPPET_OIDS
```

Comment:
We could make the same argument for this as for the csr attributes yaml file.  We duplicated the logic for parsing it in Puppet Server Clojure code even though Ruby-level Puppet has its own implementation as well.

It seems like there will continue to be a need for Ruby-level Puppet to have to put together an array of the trusted oids - built-in Puppet ones + user-defined ones - until such time as we'd be able to eliminate any of the Ruby-level code paths for certificate requests (even from the agent side).

That said, though, having a single path in the Clojure code to parse the complete list of built-in + custom oid -> name mappings and be able to inject that into both tk-auth and down to the Puppet Server certificate class in Ruby seems like it could be a good thing - even if a shadow Ruby Puppet side implementation continues to exist to support agent / certificate requests for some period of time.  We could also try to consolidate everything into a Ruby Puppet level implementation that the Clojure services mine data from but I'm not sure that's worth it if we're trying to peel more of this code away from Ruby Puppet in the long term.